luminary/Services/Filesystem/App/Storage.php

```diff
@@ -22,4 +23,67 @@ protected static function getStorage()
     {
         return app('files');
     }
+
+    /**
+     * Create a directory within the filesystem
+     * recursively w/default permissions
+     *
+     * @param string|array $path
+     * @param bool $recursive
+     * @return bool
+     */
+    public static function makeDirectory($path, bool $recursive = false) :bool
+    {
+        if (is_array($path)) {
+            foreach ($path as &$p) {
+                static::makeDirectory($p, $recursive);
+            }
+
+            return true;
+        }
+
+        return static::getStorage()->makeDirectory($path, 0755, true, $recursive);
+    }
+
+    /**
+     * Create a .gitkeep file
+     *
+     * @param $path
+     * @return string
+     */
+    public static function gitKeep($path) :string
+    {
+        $path = $path . '/.gitkeep';
+
+        if (!static::exists($path)) {
+            static::put($path, "");
+        }
+
+        return $path;
+    }
+
+    /**
+     * Create a relative link to the target file or directory.
+     *
+     * @param  string  $target
+     * @param  string  $link
+     * @return string
+     */
+    public static function link($target, $link) :string
+    {
+        if (! is_link($target)) {
```

Comment:
Saw this in a few other spots. Do we add space after logical operator or is this an oopsi?